[DeviceMesh][2D] Use concatenate for 2D (FSDP+TP) instead of getting from root mesh

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #165492
* #163358
* #166264

With concatenate API, we can directly combine two meshes together rather than getting the spmd mesh from root.

cc @H-Huang @awgu @wanchaol @fegin @wz337 @wconstab @d4l3k @pragupta @msaroufim @dcci

Differential Revision: [D85409698](https://our.internmc.facebook.com/intern/diff/D85409698)

fduwjj committed:
[DeviceMesh][2D] Use concatenate for 2D (FSDP+TP) instead of getting from root mesh

[ghstack-poisoned]

fegin reviewed (approved):
Very nice, let's wait for all the tests and can you also import this PR so that we can also trigger all the internal tests?

fduwjj committed:
Update on "[DeviceMesh][2D] Use concatenate for 2D (FSDP+TP) instead of getting from root mesh"


With concatenate API, we can directly combine two meshes together rather than getting the spmd mesh from root.

cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

fduwjj committed:
Update on "[DeviceMesh][2D] Use concatenate for 2D (FSDP+TP) instead of getting from root mesh"


With concatenate API, we can directly combine two meshes together rather than getting the spmd mesh from root.

cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

fduwjj committed:
Update on "[DeviceMesh][2D] Use concatenate for 2D (FSDP+TP) instead of getting from root mesh"


With concatenate API, we can directly combine two meshes together rather than getting the spmd mesh from root.

cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

fduwjj committed:
Update on "[DeviceMesh][2D] Use concatenate for 2D (FSDP+TP) instead of getting from root mesh"


With concatenate API, we can directly combine two meshes together rather than getting the spmd mesh from root.

cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

fduwjj committed:
Update on "[DeviceMesh][2D] Use concatenate for 2D (FSDP+TP) instead of getting from root mesh"


With concatenate API, we can directly combine two meshes together rather than getting the spmd mesh from root.

cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

fduwjj committed:
Update on "[DeviceMesh][2D] Use concatenate for 2D (FSDP+TP) instead of getting from root mesh"


With concatenate API, we can directly combine two meshes together rather than getting the spmd mesh from root.

cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

fduwjj committed:
Update on "[DeviceMesh][2D] Use concatenate for 2D (FSDP+TP) instead of getting from root mesh"


With concatenate API, we can directly combine two meshes together rather than getting the spmd mesh from root.

cc H-Huang awgu wanchaol fegin wz337 wconstab d4l3k pragupta msaroufim dcci

Differential Revision: [D85409698](https://our.internmc.facebook.com/intern/diff/D85409698)

[ghstack-poisoned]
